Tests/MeiliSearchIntegrationTests/SearchTests.swift

```diff
@@ -519,126 +534,163 @@ class SearchTests: XCTestCase {
       sortableAttributes: ["id"]
       )
 
-    let expectation = XCTestExpectation(description: "Configure filterable attributes")
-
     self.index.updateSettings(settings) { result in
       switch result {
-      case .success(let update):
-        waitForPendingUpdate(self.client, self.uid, update) {
-          expectation.fulfill()
-          completion()
+      case .success(let task):
+        self.client.waitForTask(task: task) { result in
+          switch result {
+          case .success:
+            completion(.success(()))
+          case .failure(let error):
+            dump(error)
+            completion(.failure(error))
+          }
         }
-      case .failure:
-        XCTFail("Failed to update the settings")
+      case .failure(let error):
+        dump(error)
+        completion(.failure(error))
       }
     }
   }
 
   func testSearchFilters() {
     let expectation = XCTestExpectation(description: "Search for Books using filter")
-    configureFilters {
-      typealias MeiliResult = Result<SearchResult<Book>, Swift.Error>
-      let limit = 5
-      let query = "french book"
-      let filter = "id = 456"
-      let parameters = SearchParameters(query: query, limit: limit, filter: filter)
-
-      self.index.search(parameters) { (result: MeiliResult) in
-        switch result {
-        case .success(let documents):
-          XCTAssertTrue(documents.query == query)
-          XCTAssertTrue(documents.limit == limit)
-          XCTAssertTrue(documents.hits.count == 1)
-          let book = documents.hits[0]
-          XCTAssertEqual(456, book.id)
-          XCTAssertEqual("Le Petit Prince", book.title)
-          XCTAssertEqual("A french book", book.comment)
-          expectation.fulfill()
-        case .failure(let error):
-          print(error)
-          XCTFail("Failed to search with testSearchFilters")
+    typealias MeiliResult = Result<SearchResult<Book>, Swift.Error>
+    let limit = 5
+    let query = "french book"
+    let filter = "id = 456"
+    let parameters = SearchParameters(query: query, limit: limit, filter: filter)
+
+    configureFilters { result in
+      switch result {
+      case .success:
+        self.index.search(parameters) { (result: MeiliResult) in
+          switch result {
+          case .success(let documents):
+            XCTAssertTrue(documents.query == query)
```

Comment:
Why not? `XCTAssertEqual(documents.query, query)` feels more comfortable to read :)